book/src/development/adding_lints.md

```diff
@@ -126,10 +126,14 @@ fn main() {
 ```
 
 Note that we are adding comment annotations with the name of our lint to mark
-lines where we expect an error. Once we have implemented our lint we can run
-`TESTNAME=foo_functions cargo uibless` to generate the `.stderr` file. If our
-lint makes use of structured suggestions then this command will also generate
-the corresponding `.fixed` file.
+lines where we expect an error. With the exception of very specific situations
+identified with a `//@check-pass` annotation, at least one error marker must be
+present in a test file for it to be accepted.
```

Comment:
```suggestion
lines where we expect an error. Except for very specific situations, at least one error marker must be
present in a test file for it to be accepted. Exceptional test can be identified with a `//@check-pass` annotation.
```

Suggestion messes up formatting, so better copy and paste it.

I would swap the wording: Mention the common case first and then describe what to do in the exceptional case.

Reply:
I would even remove the `//@check-pass` mention, this is too specialized for this section, and only apply when you have, for example, `ui-toml` tests with specific configurations.

Reply:
I think also for a bunch of `crashes/ice-1234.rs` tests, I think.

But yeah, not really necessary.